### ISSUE TITLE
Placeholder expansion is not working

### ISSUE THREAD

**Issue by sparrowlegs** [micronaut-projects/micronaut-openapi#1143]
### Expected Behavior

The placeholder values (prefixed with `micronaut.openapi.expand.`) in the `openapi.properties` file should be expanded into the generated YAML resource, like so:

```YAML
openapi: 3.0.1
info:
  title: broken-micronaut-openapi-expand
  description: "monkey"
  version: "2.2.2"
``` 

### Actual Behaviour

The placeholder values are not expanded in the generated YAML resource file:

```YAML
openapi: 3.0.1
info:
  title: broken-micronaut-openapi-expand
  description: "${another.placeholder.value}"
  version: "${api.version}"
``` 

### Steps To Reproduce

1. Clone the [example code repo](https://github.com/sparrowlegs/broken-micronaut-openapi-expand)
2. Change to the project directory 
3. Run `./mvnw verify`
4. The unit test failure demonstrates that the placeholder values are not expand properly

Notice in the build output that the placeholders are correctly identified (line 2). Also notice that the `api.version` placeholder value is correctly expanded into the YAML resource file name on line 3.

```
[INFO] Generating OpenAPI Documentation
[INFO] Expanding properties: [${api.version}=2.2.2, ${another.placeholder.value}=monkey]
[INFO] Writing OpenAPI file to destination: ./target/classes/META-INF/swagger/broken-micronaut-openapi-expand-2.2.2.yml
[INFO] Writing OpenAPI views to destination: ./target/classes/META-INF/swagger/views
```

5. Inspect the contents of the generated resource file: `./target/classes/META-INF/swagger/broken-micronaut-openapi-expand-2.2.2.yml`. Notice that the placeholder values are not expanded.

```YAML
openapi: 3.0.1
info:
  title: broken-micronaut-openapi-expand
  description: "${another.placeholder.value}"
  version: "${api.version}"
``` 

### Environment Information

- macOS 13.4.1 (c)
- OpenJDK 64-Bit Server VM Zulu17.42+19-CA (build 17.0.7+7-LTS, mixed mode, sharing)
- Maven 3.9.3
- Micronaut 4.0.1
- micronaut-openapi 5.0.1

### Example Application

https://github.com/sparrowlegs/broken-micronaut-openapi-expand

### Version

4.0.1

**Comment by olamelin** [comment]
We are experiencing the same issue. I noticed the commits that you added @altro3 is there a plan when they are to be merged?

**Comment by altro3** [comment]
@olamelin Hi! I can only watch the project on the weekend. I need to take a closer look at a ksp issue (https://github.com/micronaut-projects/micronaut-openapi/issues/1154) and would like to improve the documentation a bit. So at best next week. I'm talking about the release date now, I'll try to merge the PR itself today

**Comment by altro3** [comment]
So, I merged it branch 4.10, but I haven't rights to merge it to master branch. Need to wait this PR: https://github.com/micronaut-projects/micronaut-openapi/pull/1150.

I hope, after this PR we can finalize release.

**Comment by sparrowlegs** [comment]
Thank you!
